src/main/java/com/conveyal/gtfs/PatternBuilder.java

```diff
@@ -0,0 +1,242 @@
+            Pattern pattern = entry.getValue();
+            LOG.debug("Batching pattern {}", pattern.pattern_id);
+            if (!patternIdsLoadedFromFile.contains(pattern.pattern_id)) {
+                // Only insert the pattern if it has not already been imported from file.
+                pattern.setStatementParameters(insertPatternStatement, true);
+                patternTracker.addBatch();
+            }
+            createPatternStops(entry.getKey(), pattern.pattern_id, patternStopsTable);
+            updateTripPatternReferences(patternForTripsFileStream, pattern);
+        }
+        // Send any remaining prepared statement calls to the database backend.
+        patternTracker.executeRemaining();
+        LOG.info("Done storing patterns and pattern stops.");
+    }
+
+    /**
+     * Create temp table for updating trips with pattern IDs to be dropped at the end of the transaction.
+     * NOTE: temp table name must NOT be prefixed with schema because temp tables are prefixed with their own
+     * connection-unique schema.
+     */
+    private PrintStream createTempPatternForTripsTable(
+        File tempPatternForTripsTextFile,
+        Statement statement
+    ) throws SQLException, IOException {
+        LOG.info("Loading via temporary text file at {}.", tempPatternForTripsTextFile.getAbsolutePath());
+        String createTempSql = String.format("create temp table %s(trip_id varchar, pattern_id varchar) on commit drop", TEMP_FILE_NAME);
+        LOG.info(createTempSql);
+        statement.execute(createTempSql);
+        return new PrintStream(new BufferedOutputStream(Files.newOutputStream(tempPatternForTripsTextFile.toPath())));
+    }
+
+    /**
+     * Update all trips on this pattern to reference this pattern's ID.
+     */
+    private void updateTripPatternReferences(PrintStream patternForTripsFileStream, Pattern pattern) {
+        // Prepare each trip in pattern to update trips table.
+        for (String tripId : pattern.associatedTrips) {
+            // Add line to temp csv file if using postgres.
+            // No need to worry about null trip IDs because the trips have already been processed.
+            String[] strings = new String[]{tripId, pattern.pattern_id};
+            // Print a new line in the standard postgres text format:
+            // https://www.postgresql.org/docs/9.1/static/sql-copy.html#AEN64380
+            patternForTripsFileStream.println(String.join("\t", strings));
+        }
+    }
+
+    /**
+     * Copy the pattern for trips text file into a table, create an index on trip IDs, and update the trips
+     * table.
+     */
+    private void updateTripPatternIds(
+        File tempPatternForTripsTextFile,
+        PrintStream patternForTripsFileStream,
+        Statement statement,
+        String tripsTableName
+    ) throws SQLException, IOException {
+
+        LOG.info("Updating trips with pattern IDs.");
+        patternForTripsFileStream.close();
```

Comment:
Why is `patternForTripsFileStream` passed to this method? The only action on it is to close it... How about using a try-with-resources in the method that calls this one?